addons/l10n_ro_edi_stock/models/stock_picking.py

```diff
@@ -0,0 +1,864 @@
+        for picking in self:
+            picking._l10n_ro_edi_stock_reset_location_data('start')
+
+    @api.onchange('l10n_ro_edi_stock_operation_type', 'l10n_ro_edi_stock_end_loc_type')
+    def _l10n_ro_edi_stock_reset_end_location_data(self):
+        for picking in self:
+            picking._l10n_ro_edi_stock_reset_location_data('end')
+
+    def _l10n_ro_edi_stock_reset_location_data(self, location: str):
+        for picking in self:
+            picking[f'l10n_ro_edi_stock_{location}_bcp'] = False
+            picking[f'l10n_ro_edi_stock_{location}_customs_office'] = False
+            picking[f'l10n_ro_edi_stock_{location}_state_id'] = False
+            picking[f'l10n_ro_edi_stock_{location}_zip'] = False
+            picking[f'l10n_ro_edi_stock_{location}_city'] = False
+            picking[f'l10n_ro_edi_stock_{location}_street'] = False
+            picking[f'l10n_ro_edi_stock_{location}_other_info'] = False
+
+    ################################################################################
+    # Compute Methods
+    ################################################################################
+
+    @api.depends('l10n_ro_edi_stock_operation_type')
+    def _compute_l10n_ro_edi_stock_available_operation_scopes(self):
+        for picking in self:
+            allowed_scopes = [c for c, dummy in OPERATION_SCOPES]
+
+            if picking.l10n_ro_edi_stock_operation_type:
+                allowed_scopes = OPERATION_TYPE_TO_ALLOWED_SCOPE_CODES.get(picking.l10n_ro_edi_stock_operation_type, ("9999",))
+
+                if picking.l10n_ro_edi_stock_operation_scope not in allowed_scopes:
+                    picking.l10n_ro_edi_stock_operation_scope = None
+            elif picking.l10n_ro_edi_stock_operation_scope:
+                picking.l10n_ro_edi_stock_operation_scope = None
+
+            picking.l10n_ro_edi_stock_available_operation_scopes = ','.join(allowed_scopes)
+
+    @api.depends('l10n_ro_edi_stock_operation_type')
+    def _compute_l10n_ro_edi_stock_available_location_types(self):
+        for picking in self:
+            picking.l10n_ro_edi_stock_available_start_loc_types = picking._l10n_ro_edi_stock_get_available_location_types('start')
+            picking.l10n_ro_edi_stock_available_end_loc_types = picking._l10n_ro_edi_stock_get_available_location_types('end')
+
+    @api.depends('l10n_ro_edi_stock_document_ids')
+    def _compute_l10n_ro_edi_stock_current_document_data(self):
+        for picking in self:
+            document = picking._l10n_ro_edi_stock_get_current_document()
+
+            picking.l10n_ro_edi_stock_state = document.state if document else False
+            picking.l10n_ro_edi_stock_document_message = document.message if document else False
+            picking.l10n_ro_edi_stock_document_load_id = document.l10n_ro_edi_stock_load_id if document else False
+            picking.l10n_ro_edi_stock_document_uit = document.l10n_ro_edi_stock_uit if document else False
```

Comment:
I don't think this is necessary. If you need more details about your document, you can see the value of those fields in the one2Many tab directly. We need to habit ppl to check this additional tab for everything related to EDIs.

However, for the state of the document, it's correct but it must be stored and be displayed on the form/list views

Reply:
that's the thing, there isn't a one2Many tab. that's why we take the all the information necessary to show on the picking form like this